UserDocs.md

```diff
@@ -55,6 +57,8 @@ Writes the in-memory object `src` to the path `dest` in S3.
 Transparently serializes some object types.
 See [Serialization](#serialization) for details.
 
+`dest` should not end in `/`. Objects that end in `/` are treated specially by some s3 tools in a way that is potentially dangerous, so it's best to avoid them.
```

Comment:
Also not the right place to document this as the API only allows `dest/` if you have `src/` and `src/` is a dir? if that's not the case we should make it the case :)